integration/framework/avro_kafka_docker_env.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"time"
+
+	"github.com/integralist/go-findroot/find"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/pkg/retry"
+	"go.uber.org/zap"
+)
+
+const (
+	healthCheckURI          = "http://127.0.0.1:18083"
+	dockerComposeFilePath   = "/docker-compose-avro.yml"
+	controllerContainerName = "ticdc_controller_1"
+	upstreamDSN             = "root@tcp(127.0.0.1:4000)/"
+	downstreamDSN           = "root@tcp(127.0.0.1:5000)/"
+)
+
+// AvroKafkaDockerEnv represents the docker-compose service defined in docker-compose-avro.yml
+type AvroKafkaDockerEnv struct {
+	dockerComposeOperator
+}
+
+// NewAvroKafkaDockerEnv creates a new AvroKafkaDockerEnv
+func NewAvroKafkaDockerEnv() *AvroKafkaDockerEnv {
+	healthChecker := func() error {
+		resp, err := http.Get(healthCheckURI)
+		if err != nil {
+			return err
+		}
+
+		if resp.Body == nil {
+			return errors.New("kafka Connect HealthCheck returns empty body")
+		}
+		defer func() { _ = resp.Body.Close() }()
+
+		bytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		m := make(map[string]interface{})
+		err = json.Unmarshal(bytes, &m)
+		if err != nil {
+			return err
+		}
+
+		healthy, ok := m["healthy"]
+		if !ok {
+			return errors.New("kafka connect healthcheck did not return health info")
+		}
+
+		if !healthy.(bool) {
+			return errors.New("kafka connect not healthy")
+		}
+
+		return nil
+	}
+
+	st, err := find.Repo()
```

Comment:
If we use this way to locate the path of docker compose file, we must run the test case from code dir, the case can't be compiled and run in other dirs